Revit_Core_Engine/Convert/Facade/FromRevit/CurtainWall.cs

```diff
@@ -43,8 +46,47 @@ public static partial class Convert
 
         public static oM.Facade.Elements.CurtainWall CurtainWallFromRevit(this Wall wall, RevitSettings settings = null, Dictionary<string, List<IBHoMObject>> refObjects = null)
         {
-            BH.Engine.Reflection.Compute.RecordError("Conversion of CurtainWalls from Revit for the Facade discipline is not yet implemented."); 
-            return null;
+            if (wall == null)
+                return null;
+
+            settings = settings.DefaultIfNull();
+
+            CurtainWall bHoMCurtainWall = refObjects.GetValue<CurtainWall>(wall.Id);
+            if (bHoMCurtainWall != null)
+                return bHoMCurtainWall;
+
+            if (wall.StackedWallOwnerId != null && wall.StackedWallOwnerId != ElementId.InvalidElementId)
+                return null;
+
+            IEnumerable<oM.Facade.Elements.Opening> curtainPanels = wall.CurtainGrid.FacadeCurtainPanels(wall.Document, settings, refObjects);
+
+            // Get external edges of whole curtain wall
+            List<FrameEdge> extEdges = new List<FrameEdge>();
+            List<IElement1D> cwEdgeCrvs = curtainPanels.ExternalEdges();
+            foreach (ICurve crv in cwEdgeCrvs)
+            {
+                FrameEdge frameEdge = new FrameEdge { Curve = crv };
+                if (frameEdge != null)
+                    extEdges.Add(frameEdge);
+            }
+
+            if (curtainPanels == null || curtainPanels.ToList().Count == 0)
+                BH.Engine.Reflection.Compute.RecordError(String.Format("Processing of panels of Revit curtain wall failed. BHoM curtain wall without location has been returned. Revit ElementId: {0}", wall.Id.IntegerValue));
+
+            bHoMCurtainWall = new CurtainWall { ExternalEdges = extEdges, Openings = curtainPanels.ToList(), Name = wall.WallType.Name };
+            if (bHoMCurtainWall == null)
```

Comment:
`bHoMCurtainWall` will never be `null` (it is being initialised one line above), so the null check is redundant and could be removed.

Reply:
Fixed in latest commit.